File: src/main/java/com/github/njuro/jard/config/security/methods/HasAuthorities.java
package com.github.njuro.jard.config.security.methods;

import com.github.njuro.jard.user.UserAuthority;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation used to protect endpoints. Logged in user has access to endpoint if and only if he/she
 * has <strong>all</strong> of the user authorities defined in {@link #value()}.
 *
 * @see UserAuthority
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface HasAuthorities {
  UserAuthority[] value();
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/config/security/methods/HasAuthorities.kt
package com.github.njuro.jard.rewrite.config.security.methods

import com.github.njuro.jard.user.UserAuthority
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.FUNCTION
import kotlin.annotation.AnnotationTarget.PROPERTY_GETTER
import kotlin.annotation.AnnotationTarget.PROPERTY_SETTER

/**
 * Annotation used to protect endpoints. Logged in user has access to endpoint if and only if he/she
 * has **all** of the user authorities defined in [.value].
 *
 * @see UserAuthority
 */
@Retention(RUNTIME)
@Target(FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
annotation class HasAuthorities(vararg val value: UserAuthority)
